ephios/core/forms/users.py

```diff
@@ -284,29 +281,35 @@ def __init__(self, *args, **kwargs):
         self.field_class = "col-md-8"
 
 
-class WorkingHourRequestForm(Form):
-    when = DateField(widget=CustomDateInput, label=_("Date"))
-    hours = DecimalField(label=_("Hours of work"), min_value=0.5)
-    reason = CharField(label=_("Occasion"))
+class WorkingHourRequestForm(ModelForm):
+    date = forms.DateField(widget=CustomDateInput)
+
+    class Meta:
+        model = WorkingHours
+        fields = ["date", "hours", "reason"]
 
     def __init__(self, *args, **kwargs):
         self.request = kwargs.pop("request")
+        self.can_grant = kwargs.pop("can_grant", False)
+        self.user = kwargs.pop("user")
         super().__init__(*args, **kwargs)
         self.helper = FormHelper(self)
         self.helper.layout = Layout(
-            Field("when"),
+            Field("date"),
             Field("hours"),
             Field("reason"),
             FormActions(
                 Submit("submit", _("Send"), css_class="float-end"),
-                AbortLink(href=reverse("core:profile")),
+                AbortLink(href=reverse("core:workinghour_detail", args=[self.user.pk]))
+                if self.can_grant
+                else AbortLink(href=reverse("core:workinghour_own")),
```

Comment:
I must say I find having the Abort Links build inside the form kind of defies the django pattern of having forms reusable. Like that kind of navigation should be part of the view, right?